api/validation.go

```diff
@@ -74,6 +74,122 @@ func (api *API) validateAndHydrateDimensions(v dataset.Version, dims []model.Dim
 	}
 }
 
+// hydrateDimensions adds additional data (id/label) to a model.Dimension, using values provided by the dataset.
+func hydrateDimensions(filterDims []model.Dimension, dims []dataset.VersionDimension) []model.Dimension {
```

Comment:
Looks like the rebase has gone very bad, a lot of long since deleted/refactored code has been reintroduced. Please refer to the latest version of this file and try to see what has gone on